reflection.md

```diff
@@ -1,18 +1,71 @@
+	[error]  required: dsls.regex.RegularExpression
+	[error]     require(n>=0)
+	[error]            ^
+    [error] expressions/src/main/scala/dsls/regex/RegularExpression.scala:32: not found: value n
+	[error]     if (n == 0) EPSILON
+	[error]         ^
+	[error] expressions/src/main/scala/dsls/regex/RegularExpression.scala:33: not found: value n
+	[error]     else if (n ==1) this
+	[error]              ^
+    [error] expressions/src/main/scala/dsls/regex/RegularExpression.scala:34: not found: value n
+	[error]     else Concat(this, apply(n-1))
+	[error]                             ^
+	However, when I remove the assume line while keeping everything else the 
+	same, everything runs smoothly. I wish the error messages were more kind,
+	as I still haven't figured out why this is happening. 
+
+
 
 ## Comment on the design of this internal DSL
 
 Write a few brief paragraphs that discuss:
-   + What works about this design? (For example, what things seem easy and
+   # What works about this design? (For example, what things seem easy and
    natural to say, using the DSL?)
-   + What doesn't work about this design? (For example, what things seem
-   cumbersome to say?)
-   + Think of a syntactic change that might make the language better. How would
-   you implement it _or_ what features of Scala would prevent you from
-   implementing it? (You don't have to write code for this part. You could say
-   "I would use literal extension to..." or "Scala's rules for valid
-   identifiers prevent...")
\ No newline at end of file
+   I believe it takes a shorter time for users to get used to using regular
+   expression operators than learning to use the existing methods. For example,
+   the operator '||' is intuitive enough as opposed to the method 'Union' because 
+   the method requres a certain number of inputs in certain ways that users might
+   not be familiar with. 
+
+   # What doesn't work about this design? (For example, what things seem
+   cumbersome to say?) Think of a syntactic change that might make the language
+   better. How would you implement it _or_ what features of Scala would prevent 
+   you from implementing it? (You don't have to write code for this part. You 
+   could say "I would use literal extension to..." or "Scala's rules for valid
+   identifiers prevent...")
+   While some regular expression operators are easy to use, there are some 
+   exceptions: the Kleene star operator (<*>) and the repetition operator ({n}, 
+   where n is the number of repetitions of the expression). They are not even
+   one-character long, which means it is as convenient for users to type as other
+   operators. Plus, I don't think they are intuitive enough. I would rather have
+   "^n" to represent as the repetition operator. Also, I would have just the star 
+   operator. I would do the same for the '+'' operator, but I see why it is between
+   the angle brackets: there's already universally understood purpose of just the 
+   '+' operator.
```

Comment:
Thats part of the fun of DSLs, though. We subvert typical usage of a language for the sake of creating a coherent new language :smiley: